address/address.go

```diff
@@ -118,6 +118,10 @@ type Tap struct {
 	// assetGen is the receiving asset's genesis metadata which directly
 	// maps to its unique ID within the Taproot Asset protocol.
 	assetGen asset.Genesis
+
+	// ProofCourierAddr is the address of the proof courier that will be used to distribute
+	// related proofs for this address.
+	ProofCourierAddr []byte
```

Comment:
I think we'd want this to be further abstracted. What's `[]byte` even mean here? Would expect a `string`, but then you'd abstract away from the actual underlying mechanism.

Reply:
I can add a more descriptive type with an enum for supported protocols. And `[]byte` serialization/deserialization.

Reply:
Oli said the following in keybase:

> I think a valid answer to the comment linked above is just: to avoid circular package dependencies the address just knows it's a string that contains an URI like scheme. how that looks and what it means is defined in the proof package (with the courier)

I agree with him and I've encountered the circular package dependency problem.